tools/myst/host/exec.c

```diff
@@ -146,13 +146,34 @@ int exec_launch_enclave(
     myst_buf_t mount_mappings_buf = MYST_BUF_INITIALIZER;
     pid_t target_tid = (pid_t)syscall(SYS_gettid);
     struct timespec start_time;
+    oe_enclave_setting_context_switchless_t switchless_setting = {0, 0};
+    oe_enclave_setting_t settings[8];
+    size_t num_settings = 0;
 
     /* get the start time and pass it into the kernel */
     if (clock_gettime(CLOCK_REALTIME, &start_time) != 0)
         _err("clock_gettime() failed");
 
+    // Initialize the switchless setting; this applies to ocalls with the
+    // transition_using_threads attribute.
+    {
+        switchless_setting.max_enclave_workers = 0;
+        switchless_setting.max_host_workers = 4;
```

Comment:
Can you use `top` to see if the `max_host_workers` makes a difference on CPU usage? We have to be conscious of not being a `CPU hog`.

Reply:
Good idea. Will check.

Reply:
Here is the peak usage for 1, 2, 4, and 8 threads respectively.

```
16535 mike      20   0  544776  19120   6180 R 207.9  0.2   0:10.90 myst [1]
16242 mike      20   0  552972  19304   6356 R 208.6  0.3   0:11.89 myst [2]
16673 mike      20   0  569364  19168   6204 R 207.9  0.2   0:11.22 myst [4]
16386 mike      20   0  602148  19128   6132 R 208.3  0.2   0:11.03 myst [8]
```

Reply:
The performance seems more or less the same no matter how many worker threads there are. What do you think is a good number?

Reply:
So the number of worker threads has an impact on memory but not CPU load. Interesting.